src/test/java/com/epam/lab/paymentsystem/service/OperationServiceTest.java

```diff
@@ -0,0 +1,204 @@
+    assertEquals(operations, operationService.getAllOperations());
+  }
+}*/
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+
+import com.epam.lab.paymentsystem.entities.Account;
+import com.epam.lab.paymentsystem.entities.Card;
+import com.epam.lab.paymentsystem.entities.Operation;
+import com.epam.lab.paymentsystem.entities.User;
+import com.epam.lab.paymentsystem.repository.OperationRepository;
+import com.epam.lab.paymentsystem.service.impl.OperationServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+@ExtendWith(MockitoExtension.class)
+public class OperationServiceTest {
+  private long accountId;
+  private String login;
+  private List<Card> cards;
+  private List<Operation> operations;
+  private User user;
+  private Card cardSrc;
+  private Card cardDst;
+  private Operation operation;
+  private Account accountSrc;
+  private Account accountDst;
+  private long cardId;
+  private long prevSrcAmount;
+  private long prevDstAmount;
+
+  @Mock
+  private OperationRepository operationRepository;
+
+  @Mock
+  private UserService userService;
+
+  @Mock
+  private AccountService accountService;
+
+  @Mock
+  private CardService cardService;
+
+  @InjectMocks
+  private OperationServiceImpl operationService;
+
+  @BeforeEach
+  public void startUp() {
+    MockitoAnnotations.initMocks(this);
+    accountId = 1;
+    cardId = 1;
+    cards = new ArrayList<>();
+    operations = new ArrayList<>();
+
+    prevSrcAmount = 500;
+    prevDstAmount = 100;
+
+    accountSrc = new Account(null, "accountSrc", prevSrcAmount, true);
+    accountDst = new Account(null, "accountDst", prevDstAmount, true);
+    cardSrc = new Card(accountSrc, null, "cardSrc", true);
+    cardSrc.setId(1);
+    cardDst = new Card(accountDst, null, "cardDst", true);
+    cardDst.setId(2);
+    operation = new Operation(cardSrc, cardDst, 200, null);
+
+    login = "test";
+    user = new User();
+    user.setLogin(login);
+  }
+
+  @Test
+  public void testGetAllOperationsReturnsOperationsList() {
+    when(userService.getCurrentUserLogin()).thenReturn(user.getLogin());
+    when(cardService.getAllCardsByLogin(user.getLogin())).thenReturn(cards);
+    when(operationRepository.getAllBySourceCardIsInOrTargetCardIsIn(cards, cards))
+        .thenReturn(operations);
+    assertEquals(operations, operationService.getAllOperations());
+  }
+
+  @Test
+  public void testMakePayment() {
+    Answer<Card> cardAnswer = invocationOnMock -> {
+      Long idCard = (Long) invocationOnMock.getArgument(0);
+      if (idCard.equals(1L)) {
```

Comment:
Can you use ternary operator here?

Reply:
I have just fixed it and made test right workable)